Feature/ecbuildification obs2ioda

rajichidamb opened the pull request:
## Description

Changing code to use eckit Yaml configuration instead of Yaml-cpp implemented by NCAR. CTest updated to compare the output file with reference file.

jedi-ci-test-select=gcc

## Issue(s) addressed

Resolves #(https://github.com/JCSDA-internal/ioda-converters/issues/1694)

## Dependencies

## Impact

Not sure.

## Checklist

- [X] I have performed a self-review of my own code
- [] I have made corresponding changes to the documentation
- [X] I have run the unit tests before creating the PR

rajichidamb committed:
First changes remove extra cmake files

rajichidamb committed:
Include FetchContent

rajichidamb committed:
Add the function commented out

rajichidamb committed:
Removing Yaml.cpp and adding eckit yaml configuration. Updating the test reference file amsua_metop-b_obs_2020110112.nc4, mhs_metop-b_obs_2020101215.nc4 and adding new reference file sondes_obs_2020093018.nc4. Removing config folder

amstokely commented:
@ibanos90 This PR changes the output NetCDF file extension from .h5 to .nc4. Will this affect any workflows that PANDA-C runs? If so, we could add a CLI option to let users specify the output file extension.

rajichidamb commented on src/ncar/obs2ioda/src/cxx/ioda_obs_schema.cc:
```diff
@@ -65,12 +82,13 @@ IodaObsDimension::IodaObsDimension(std::string name)
         : IodaObsSchemaComponent("Variable", std::move(name)) {}
 
-void IodaObsVariable::load(const std::shared_ptr<IYamlNode> &node) {
+void IodaObsVariable::load(const std::shared_ptr<IYamlNode> &node,
+                           const eckit::LocalConfiguration &config) {
```
Oh, no missed the change, will fix.

rajichidamb commented on src/ncar/obs2ioda/src/cxx/ioda_obs_schema.h:
```diff
@@ -8,7 +8,9 @@
+#include "eckit/config/YAMLConfiguration.h"
+#include "eckit/config/LocalConfiguration.h"
+#include "eckit/filesystem/PathName.h"
 #include "FilePathConfig.h"
```
Hm, I forgot this, let me take a look, Thanks

ibanos90 commented:
> @ibanos90 This PR changes the output NetCDF file extension from .h5 to .nc4. Will this affect any workflows that PANDA-C runs? If so, we could add a CLI option to let users specify the output file extension.

Hi @amstokely, thank you for tagging me here!

I’m a bit confused, is there a specific reason why this PR changes the file extension? Is this related to something new in the IODAv3 format (apologies if I missed something)? Is there a follow-up development planned to revert the extension back to `.h5`?

For our PANDAC runs, we use files in the IODAv3 format, which have the `.h5` extension. Since we use pre-generated files, our current runs are unlikely to break, but any processing for a new period using an updated obs2ioda executable will likely fail.

BenjaminRuston commented:
> I’m a bit confused, is there a specific reason why this PR changes the file extension? Is this related to something new in the IODAv3 format (apologies if I missed something)? Is there a follow-up development planned to revert the extension back to .h5?

@amstokely and @ibanos90 the convention by most groups apparently unspoken was either `.nc` or `.nc4` so using `.h5` is the new introduction.  I know r2d2 has a field for extension so you can use all 3 seamlessly

I like @ibanos90 proposal to be able to set this.

ibanos90 commented:
> > I’m a bit confused, is there a specific reason why this PR changes the file extension? Is this related to something new in the IODAv3 format (apologies if I missed something)? Is there a follow-up development planned to revert the extension back to .h5?
> 
> @amstokely and @ibanos90 the convention by most groups apparently unspoken was either `.nc` or `.nc4` so using `.h5` is the new introduction. I know r2d2 has a field for extension so you can use all 3 seamlessly
> 
> I like @ibanos90 proposal to be able to set this.

Oh, I see. Thanks for the clarification, @BenjaminRuston!

I agree, @amstokely's idea for setting this is would be great!

rajichidamb commented:
> > > I’m a bit confused, is there a specific reason why this PR changes the file extension? Is this related to something new in the IODAv3 format (apologies if I missed something)? Is there a follow-up development planned to revert the extension back to .h5?
> > 
> > 
> > @amstokely and @ibanos90 the convention by most groups apparently unspoken was either `.nc` or `.nc4` so using `.h5` is the new introduction. I know r2d2 has a field for extension so you can use all 3 seamlessly
> > I like @ibanos90 proposal to be able to set this.
> 
> Oh, I see. Thanks for the clarification, @BenjaminRuston!
> 
> I agree, @amstokely's idea for setting this is would be great!

Since ctest was expecting nc4 file, and the ncar-bufr2nc-fortran codebase was writing out file with .nc4 extension, I made the change. I also see the  name_var_info in define_mod.f90 has also changed, it is missing LaunchTime and has added datetime any specific reason for this?

BenjaminRuston commented:
think we need both.... `dateTime` is per location in the vertical while the `LaunchTime` is fixed

so we actually need both

rajichidamb committed:
Adding the config folder and new cmake/iodaconv_functions.cmake to include the functions for setting targets and files. Fixing the load function parameters

rajichidamb commented on src/ncar/obs2ioda/src/cxx/netcdf_file.cc:
```diff
@@ -4,8 +4,9 @@
 namespace Obs2Ioda {
-    auto iodaSchemaYamlRoot = std::make_shared<YamlCppNode>(
-            YAML::LoadFile(IODA_SCHEMA_YAML));
+    std::string yamlPath = IODA_SCHEMA_YAML;
```
Added the config folder to fix this issue.

srherbener commented on src/ncar/obs2ioda/src/CMakeLists.txt:
```diff
@@ -25,15 +25,19 @@ set(obs2ioda_v3_SOURCES
 )
-set(v3_PUBLIC_LINK_LIBRARIES NetCDF::NetCDF_Fortran bufr::bufr_4 obs2ioda_cxx)
+set(v3_PUBLIC_LINK_LIBRARIES eckit NetCDF::NetCDF_Fortran bufr::bufr_4 obs2ioda_cxx)
 add_library(v3 SHARED ${v3_SOURCES})
```
I'm not crazy about a library named "v3" (libv3.so). It's not very descriptive and could collide with other libraries. Would it be feasible to change this name? If so, would something like "obs2ioda_utils" work?

rajichidamb replied:
How about obs2ioda_v3utils, since the target is obs2ioda_v3

rajichidamb replied:
Updated to obs2ioda_utils and obs2ioda as target name

rajichidamb commented:
> think we need both.... `dateTime` is per location in the vertical while the `LaunchTime` is fixed
> 
> so we actually need bot

> think we need both.... `dateTime` is per location in the vertical while the `LaunchTime` is fixed
> 
> so we actually need both

There is dateTime and datetime, but missing LaunchTime

srherbener commented on test/CMakeLists.txt:
```diff
@@ -2317,15 +2324,18 @@ if( iodaconv_pbfortran_ENABLED )
-                            -i testinput -o testrun gdas.t18z.1bmhs.tm00.bufr_d"
-                            mhs_metop-b_obs_2020101215.nc4 ${IODA_CONV_COMP_TOL}
-                    DEPENDS bufr2nc_fortran.x)
+#   ecbuild_add_test( TARGET  test_${PROJECT_NAME}_mhs_conv
```
I'm assuming that the plan is to evetually obsolete the bufr2nc_fortran.x converter, and that this will take some coordination (ie, time) to migrate the current users of bufr2nc_fortran.x to obs2ioda_v3.

It also appears that a radiosonde_obs_202093018.nc4 file (used in the remaining prepbufr2nc_fortran.x test) was left in the ioda v1 format to prevent collision with the ioda v3 sondes_obs_... file.

If this is correct, it might be useful to continue testing bufr2nc_fortran.x with MHS and AMSUA obs types. Would it be feasible to do what was done for sondes (have both v1 and v3 test reference files) so that the MHS and AMSUA tests can be enabled?

If this doesn't seem like a good approach to you, (perhaps the obsolescence of prepbufr2nc_fortran.x can take place quickly, or you don't want to increase the size of the git-lfs portion of the repo, etc) then I think these tests should be removed completely and not left as commented out configuration.

rajichidamb replied:
Since the converted file written to testrun has same name but different format for bufr2nc_fortran.x to obs2ioda_v3, the testoutput file needs to have different reference file for comparison. I can write a different script which can take testrun and testoutput file with different name and keep both the tests.

rajichidamb replied:
Created a script to pass in a different names for test run and test output file name and compare.

srherbener reviewed (commented):
This looks great, just a couple minor comments. Thanks for making this extremely useful change!

srherbener commented:
> > think we need both.... `dateTime` is per location in the vertical while the `LaunchTime` is fixed
> > so we actually need bot
> 
> > think we need both.... `dateTime` is per location in the vertical while the `LaunchTime` is fixed
> > so we actually need both
> 
> There is dateTime and datetime, but missing LaunchTime

I agree that LaunchTime is needed. It is used to keep track of multiple balloon launches (ie, keep them separate) at the same station. The obs grouping feature only uses the variable(s) given to it to decide how to form the groupings, so if only station ID is used, multiple ballon launches will get grouped together. If station ID and LaunchTime are both used, then the separate profiles (from the multiple launches) will be placed in separate groupings. 

datetime is the deprecated ISO 8601 string format which is being ignored by the ioda reader and thus should be removed.

rajichidamb committed:
New comparator file for nc4 specific for bufr2nc_fortran converter

rajichidamb committed:
Add launchTime and remove datetime. This causes the min_datetime and max_datetime string to be null, so modifying the testoutput files too

BenjaminRuston commented:
thank you @srherbener for the useful comments also note the `LaunchTime` can be used to identify duplicates of the same report maybe made in a high resolution (time and location - raw format from new GNSS balloons) and a lower resolution (TAC or PrepBUFR)

This is very late to be saying this but I'm not crazy about the name `obs2ioda_v3` as right after this work is complete will likely have some bug fixes upcoming.. so rather than `v3.0.1` ... `v.3.1.4` etc. would we want `obs2ioda_utils` and just `obs2ioda.x` does that currently clash with anything?

may be off on this will have a think and catch @rajichidamb and/or you @srherbener and continue to move this along. thx

eap committed:
Migrate ioda-converters to jedi-ci action (#1696)

Add jedi-ci action

Parent issue: JCSDA-internal/jedi-ci#16
Admin merge justification: JCSDA-internal/jedi-ci#17

rajichidamb committed:
Merge branch 'develop' into feature/ecbuildification_obs2ioda

srherbener commented:
I like @BenjaminRuston's suggestions for names (obs2ioda, obs2ioda_utils). It does seem troublesome including a version number in the application and library names.

srherbener commented on .gitattributes:
```diff
@@ -1,5 +1,5 @@
 *.he5 filter=lfs diff=lfs merge=lfs -text
-src/ncar/obs2ioda/test/validation/data/goes_abi/write_ioda_v3/reference/write_goes_abi_ioda_v3.nc  -filter -diff -merge
+#src/ncar/obs2ioda/test/validation/data/goes_abi/write_ioda_v3/reference/write_goes_abi_ioda_v3.nc  -filter -diff -merge
```
Can this line be removed? I'm not clear on why it is included so it may need to remain, but I think it should be removed if it serves no purpose.

rajichidamb replied:
Removed.

srherbener commented on test/CMakeLists.txt:
```diff
@@ -2294,17 +2297,23 @@ ecbuild_add_test( TARGET  test_${PROJECT_NAME}_osw_cygnss
+                            netcdf
+                            "${CMAKE_BINARY_DIR}/bin/obs2ioda_v3
+                            -i testinput -o testrun prepbufr.bufr"
+                            sondes_obs_2020093018.nc4 ${IODA_CONV_COMP_TOL}
```
Thanks for addressing the prepbufr2nc_fortran.x and obs2ioda test file issues! I think what you've done is good but it needs one more step. We should support running ctests in parallel (eg, ctest -j10) and since there are pairs of tests that write to the same file (eg, test_iodaconv_prepbufr_mhs_conv_ncar_obs2ioda and test_iodaconv_mhs_conv both write to mhs_metop-b_obs_202101215.nc4) file collisions can occur and cause test failures. The way to fix this is to add `TEST_DEPENDS <testname>` arguments to the ecbuild_add_test commands. If you make the prepbufr2nc_fortran.x tests depend on their obs2ioda couterparts, then this forces the obs2ioda test to complete before running the prepbufr2nc_fortran.x test (thus preventing the file collision). Making the dependency go this way also makes it easy to remove the prepbufr2nc_fortran.x tests when that application is deprecated.

Please add the appropriate TEST_DEPENDS to the mhs and amsua prepbufr2nc_fortran.x tests. Thanks!

rajichidamb replied:
Added, thanks, this is awesome.

srherbener commented on tools/iodaconv_v1comp.sh:
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
```
Thanks for making this change. I think I understand why there is a new compare script - to avoid having to run through many tests and update the arguments to the tests using the current compare script. I think this is a good choice and it makes it easier to clean up the soon to be obsolete prepbufr2nc_fortran.x tests (once that application is deprecated). No need to change anything here.

For a subsequent PR it might be good to get rid of the hard coded output file names in obs2ioda. Instead give obs2ioda an obs type, an input file name, and an output file name. Then write/modify wrapper scripts to do the file name patterns outside of the obs2ioda application. This would help address the issue with the test files that we hit in this PR, and it's much easier if it becomes desirable to modify file name patterns or use different file name patterns in different use cases, etc.

srherbener reviewed (changes requested):
Looks great! Just have the issue with support for running parallel ctests.

rajichidamb committed:
Adding test dependency for parallel run of Ctests. Fix the library and target name for cmake, remove the v3 version.

rajichidamb committed:
Adding the script iodaconv_v1comp.sh to CMake list and fix the error message in the script

rajichidamb committed:
Fix the fortran cmake flags, fix the write_obs function to take file extension from command line (only h5/nc4). Change one test to pass the file extension, Change README.md file to match the updated target name without version

rajichidamb commented:
> > > I’m a bit confused, is there a specific reason why this PR changes the file extension? Is this related to something new in the IODAv3 format (apologies if I missed something)? Is there a follow-up development planned to revert the extension back to .h5?
> > 
> > 
> > @amstokely and @ibanos90 the convention by most groups apparently unspoken was either `.nc` or `.nc4` so using `.h5` is the new introduction. I know r2d2 has a field for extension so you can use all 3 seamlessly
> > I like @ibanos90 proposal to be able to set this.
> 
> Oh, I see. Thanks for the clarification, @BenjaminRuston!
> 
> I agree, @amstokely's idea for setting this is would be great!

Added command line option to pass the file extension. But for now only accepting nc4 or h5 and defaulting to nc4. Just looking at the comment 'nc' is also valid name it seems, hm. what are the suggestions?

BenjaminRuston commented:
@srherbener can you please re-review when you have a moment believe this is ready

@amstokely can we lock this in and then attack the either option to use yaml-cpp or potentially backward compatibility with earlier eckit a bit confused why it wouldn't work with a recent but older version of eckit....

if possible would like to separate this last PANDAS-C specific issue

BenjaminRuston committed:
trigger CI

srherbener commented on test/CMakeLists.txt:
```diff
@@ -2316,16 +2325,19 @@ if( iodaconv_pbfortran_ENABLED )
                             -i testinput -o testrun prepbufr.bufr"
                             radiosonde_obs_2020093018.nc4 ${IODA_CONV_COMP_TOL}
                     DEPENDS bufr2nc_fortran.x)
+                    #TEST_DEPENDS test_${PROJECT_NAME}_prepbufr_conv_ncar_obs2ioda)
```
It would be good to remove this commented out line. Thanks!

rajichidamb replied:
removed.

srherbener reviewed (approved):
Thanks for addressing my comments!

rajichidamb committed:
Remove the commented out line

rajichidamb committed:
Trying to fix the GCC build error relocation truncated to fit: R_X86_64_PC32 against symbol

BenjaminRuston commented:
thanks @srherbener 

@amstokely are you OK with moving forward and then see if we can address the backward compatibility,,, hopefully if isolated to just this we can get the older eckit to work (can we build with your stack on Derecho?)

BenjaminRuston reviewed (approved):
thanks @rajichidamb and @amstokely I ran for a group of observations, `cris` and `mhs` and then `prepBUFR` which exports `adpsfc` `aircraft` `ascat` `radiosonde` `satwind` and `sfcshp`

doing an `nccmp` only found difference in the cris_n21 surprisingly

this is not worrysome and more likely due to the CRTM coefficient used in the application of the plank function (converting radiance to brightness temperature)

in short this looks good to go and ready to be used as the new baseline `obs2ioda`

BenjaminRuston commented:
just FYI, here's the cris-fsr_n21 differences:

```
obs.20250831T210000Z_PT6H_cris-fsr_n21.nc4
DIFFER : VARIABLE : sensorCentralWavenumber : POSITION : [0] : VALUES : 7.20199e-37 <> 5.29256e-07
DIFFER : VARIABLE : sensorCentralWavenumber : POSITION : [2] : VALUES : 7.20199e-37 <> 5.29256e-07
DIFFER : VARIABLE : sensorCentralWavenumber : POSITION : [4] : VALUES : -3.93032e+10 <> -4.45594e+15
DIFFER : VARIABLE : sensorCentralWavenumber : POSITION : [6] : VALUES : -3.93032e+10 <> -4.45594e+15
Variable                Group     Count         Sum      AbsSum          Min        Max      Range        Mean      StdDev
sensorCentralWavenumber /MetaData     4 8.91181e+15 8.91181e+15 -5.29256e-07 4.4559e+15 4.4559e+15 2.22795e+15 2.57262e+15
```

BenjaminRuston commented:
yes looking into this a bit more.. the `cris-fsr_n20` looks correct in the *old* version and this *new* one:
```
>>> g['MetaData']['sensorCentralWavenumber'][:]
array([ 661.25 ,  664.375,  665.625,  666.25 ,  666.875,  668.75 ,
        669.375,  670.   ,  672.5  ,  673.75 ,  675.625,  676.875,
        678.75 ,  680.   ,  680.625,  681.25 ,  681.875,  682.5  ,
        683.125,  683.75 ,  684.375,  685.   ,  685.625,  686.25 ,
        686.875,  687.5  ,  688.125,  688.75 ,  689.375,  690.   ,
        690.625,  691.25 ,  691.875,  692.5  ,  693.125,  693.75 ,
        694.375,  695.   ,  695.625,  696.25 ,  696.875,  697.5  ,
        698.125,  698.75 ,  699.375,  700.   ,  700.625,  701.25 ,
        701.875,  702.5  ,  703.125,  703.75 ,  704.375,  705.   ,
        705.625,  706.25 ,  706.875,  707.5  ,  708.125,  708.75 ,
        709.375,  710.   ,  710.625,  711.25 ,  711.875,  712.5  ,
        713.125,  713.75 ,  714.375,  715.   ,  715.625,  716.25 ,
        716.875,  717.5  ,  718.125,  718.75 ,  719.375,  720.   ,
        720.625,  721.25 ,  721.875,  722.5  ,  723.125,  723.75 ,
        724.375,  725.   ,  725.625,  726.25 ,  726.875,  727.5  ,
        728.125,  728.75 ,  729.375,  730.   ,  730.625,  731.25 ,
        731.875,  732.5  ,  733.125,  733.75 ,  734.375,  735.   ,
        735.625,  736.25 ,  736.875,  737.5  ,  738.125,  738.75 ,
        739.375,  740.   ,  740.625,  741.25 ,  741.875,  742.5  ,
        743.125,  743.75 ,  744.375,  745.   ,  745.625,  746.25 ,
        746.875,  747.5  ,  748.125,  748.75 ,  749.375,  750.   ,
        750.625,  751.25 ,  751.875,  752.5  ,  753.125,  753.75 ,
        754.375,  755.   ,  755.625,  756.25 ,  756.875,  757.5  ,
        758.125,  758.75 ,  759.375,  760.   ,  760.625,  761.25 ,
        761.875,  762.5  ,  763.125,  763.75 ,  764.375,  765.   ,
        765.625,  766.25 ,  766.875,  767.5  ,  768.125,  768.75 ,
        769.375,  770.   ,  770.625,  771.25 ,  771.875,  772.5  ,
        773.125,  773.75 ,  774.375,  779.375,  781.25 ,  784.375,
        789.375,  795.625,  796.875,  798.125,  798.75 ,  800.625,
        803.125,  804.375,  808.75 ,  814.375,  815.625,  816.875,
        821.25 ,  823.75 ,  826.25 ,  827.5  ,  831.25 ,  833.75 ,
        837.5  ,  840.   ,  843.75 ,  856.875,  863.125,  892.5  ,
        899.375,  900.625,  901.875,  903.125,  905.625,  916.25 ,
        923.75 ,  924.375,  925.   ,  927.5  ,  930.   ,  933.75 ,
        935.625,  937.5  ,  939.375,  941.25 ,  943.125,  945.   ,
        946.25 ,  950.625,  953.125,  953.75 ,  955.625,  957.5  ,
        959.375,  961.25 ,  962.5  ,  963.75 ,  965.   ,  968.75 ,
        970.   ,  970.625,  973.125,  973.75 ,  974.375,  975.625,
        980.   ,  983.125, 1001.25 , 1004.375, 1008.75 , 1019.375,
       1020.625, 1021.875, 1023.125, 1024.375, 1025.625, 1026.875,
       1031.25 , 1033.125, 1034.375, 1035.625, 1036.875, 1038.125,
       1040.625, 1043.75 , 1048.125, 1053.125, 1054.375, 1056.875,
       1061.25 , 1070.   , 1071.25 , 1073.125, 1076.875, 1079.375,
       1083.125, 1086.875, 1091.25 , 1093.125, 1095.   , 1210.   ,
       1212.5  , 1213.75 , 1215.   , 1216.875, 1218.75 , 1223.125,
       1227.5  , 1231.25 , 1234.375, 1240.   , 1251.25 , 1253.75 ,
       1262.5  , 1294.375, 1301.25 , 1302.5  , 1305.   , 1310.   ,
       1315.   , 1320.   , 1325.   , 1330.   , 1330.625, 1331.25 ,
       1335.   , 1349.375, 1371.25 , 1371.875, 1375.   , 1376.25 ,
       1376.875, 1381.25 , 1385.625, 1387.5  , 1388.75 , 1390.625,
       1393.75 , 1394.375, 1395.   , 1397.5  , 1399.375, 1400.   ,
       1401.25 , 1402.5  , 1403.75 , 1405.   , 1406.875, 1407.5  ,
       1408.75 , 1410.   , 1411.875, 1412.5  , 1414.375, 1415.   ,
       1416.25 , 1417.5  , 1419.375, 1420.   , 1421.875, 1422.5  ,
       1425.   , 1426.25 , 1427.5  , 1428.75 , 1430.   , 1431.875,
       1436.25 , 1436.875, 1438.75 , 1442.5  , 1445.625, 1448.125,
       1451.875, 1456.875, 1458.75 , 1464.375, 1468.75 , 1471.875,
       1490.625, 1496.25 , 1505.625, 1506.875, 1516.875, 1520.625,
       1525.625, 1533.125, 1541.875, 1558.125, 1569.375, 1576.25 ,
       1584.375, 1586.25 , 1591.875, 1594.375, 1605.   , 1605.625,
       1684.375, 1685.   , 1695.625, 1700.625, 1734.375, 1745.   ,
       2165.625, 2169.375, 2180.   , 2183.125, 2190.   , 2380.   ,
       2380.625, 2381.25 , 2381.875, 2382.5  , 2383.125, 2383.75 ,
       2384.375, 2385.   , 2385.625, 2386.25 , 2386.875, 2387.5  ,
       2388.125, 2388.75 , 2389.375, 2390.   , 2390.625, 2391.25 ,
       2391.875, 2392.5  , 2393.125, 2393.75 , 2394.375, 2395.   ,
       2395.625, 2396.25 , 2396.875, 2397.5  , 2398.125, 2398.75 ,
       2399.375, 2400.   , 2400.625, 2401.25 , 2401.875, 2402.5  ,
       2403.125, 2403.75 , 2404.375, 2405.   , 2405.625, 2406.25 ,
       2406.875, 2407.5  , 2408.125, 2408.75 , 2409.375, 2410.   ,
       2492.5  , 2505.625, 2507.5  , 2510.   , 2513.75 , 2516.875,
       2518.75 , 2523.125, 2525.   , 2527.5  , 2531.875], dtype=float32)

```



while both the old and new for `cris-fsr_n21` have an array that is populated with `0.0` except for the first 8 positions... so basically it was not correct in either the old or the new version and needs to be corrected (fixed).

Here's the *new* `obs2ioda`:
```
>>> f['MetaData']['sensorCentralWavenumber'][:]
array([ 7.2019907e-37,  2.2958874e-41,  7.2019907e-37,  2.2958874e-41,
       -3.9303250e+10,  6.1222730e-41, -3.9303250e+10,  6.1222730e-41,
        0.0000000e+00,  0.0000000e+00,  0.0000000e+00,  0.0000000e+00,
        ...
        0.0000000e+00,  0.0000000e+00,  0.0000000e+00], dtype=float32)
```

and here's the *old* `obs2ioda`:
```
>>> g['MetaData']['sensorCentralWavenumber'][:]
array([ 5.2925589e-07,  2.2958874e-41,  5.2925589e-07,  2.2958874e-41,
       -4.4559427e+15,  6.1222730e-41, -4.4559427e+15,  6.1222730e-41,
        0.0000000e+00,  0.0000000e+00,  0.0000000e+00,  0.0000000e+00,
        ...
        0.0000000e+00,  0.0000000e+00,  0.0000000e+00], dtype=float32)
```

so essentially both have random garbage in the `MetaData` for `sensorCentralWavenumber` for `cris-fsr_n21`

BenjaminRuston committed:
Merge branch 'develop' into feature/ecbuildification_obs2ioda

ashley314 reviewed (approved):
Thanks for this change! It will greatly simplify our conversion in the ingest suite. I opened a ticket to update those scripts after this get's merged. https://github.com/JCSDA-internal/ewok/issues/1210.
